### PR TITLE
Add L2 test for LAG hash seed

### PR DESCRIPTION
**Pull request by yuriilisovskyi**

### CODE REVIEW

File: tests/test_l2_basic.py [hunk 1]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
+
+    # Create VLAN
+    vlan_oid = npu.create(SaiObjType.VLAN, ["SAI_VLAN_ATTR_VLAN_ID", vlan_id])
+
+    # Create VLAN members
+    npu.create_vlan_member(vlan_oid, lag_bp_oid, "SAI_VLAN_TAGGING_MODE_UNTAGGED")
+    npu.create_vlan_member(vlan_oid, npu.dot1q_bp_oids[4], "SAI_VLAN_TAGGING_MODE_UNTAGGED")
+
+    # Set PVID for LAG and Port #4
+    npu.set(npu.port_oids[4], ["SAI_PORT_ATTR_PORT_VLAN_ID", vlan_id])
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:441]: Let's use `max_port` here.

File: tests/test_l2_basic.py [hunk 2]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
+                exp_pkt = simple_tcp_packet(eth_dst=mac,
+                                            eth_src=src_mac,
+                                            ip_dst=ip_dst,
+                                            ip_src=ip_src,
+                                            tcp_sport=sport,
+                                            tcp_dport=dport,
+                                            ip_id=109,
+                                            ip_ttl=64)
+
+                send_packet(dataplane, 4, str(pkt))
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:481]: Let's use `max_port` here.

File: tests/test_l2_basic.py [hunk 3]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:523]: Let's use `max_port` here.

File: tests/test_l2_basic.py [hunk 4]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
+            
+            assert order_check > 0, "Checking the difference in order/sequence before and after changing hash seed value: " + str(order_check)
+
+    finally:
+        npu.flush_fdb_entries(npu.switch_oid, ["SAI_FDB_FLUSH_ATTR_BV_ID", vlan_oid, "SAI_FDB_FLUSH_ATTR_ENTRY_TYPE", "SAI_FDB_FLUSH_ENTRY_TYPE_ALL"])
+
+        npu.set(npu.switch_oid, ["SAI_SWITCH_ATTR_LAG_DEFAULT_HASH_SEED", "0"])
+
+        npu.remove_vlan_member(vlan_oid, lag_bp_oid)
+        npu.remove_vlan_member(vlan_oid, npu.dot1q_bp_oids[4])
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:543]: Let's use `max_port` here.

File: tests/test_l2_basic.py [hunk 5]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
+                                    "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE", "true"
+                                ])
+            npu.dot1q_bp_oids[idx] = bp_oid
+
+        # Add ports to default VLAN
+        for oid in npu.dot1q_bp_oids[0:5]:
+            npu.create_vlan_member(npu.default_vlan_oid, oid, "SAI_VLAN_TAGGING_MODE_UNTAGGED")
+
+        # Set PVID
+        for oid in npu.port_oids[0:5]:
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:568]: same here

File: tests/test_l2_basic.py (hunk 1 above)
Comment (andriy-kokhan) [reply tests/test_l2_basic.py:441]: or probably we can rename `max_port` to something like `lag_mbr_num`

File: tests/test_l2_basic.py [hunk 6]
```diff
@@ -383,6 +383,192 @@ def test_l2_lag(npu, dataplane):
...
+    # Remove bridge ports
+    for idx in range(max_port):
+        npu.remove_vlan_member(npu.default_vlan_oid, npu.dot1q_bp_oids[idx])
+        npu.remove(npu.dot1q_bp_oids[idx])
+
+    # Remove Port #4 from the default VLAN
+    npu.remove_vlan_member(npu.default_vlan_oid, npu.dot1q_bp_oids[4])
+
+    # Create LAG
+    lag_oid = npu.create(SaiObjType.LAG, [])
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:413]: Hm.. should we make `def create(..., attrs=[], .. )`? what do you thing?

Comment (yuriilisovskyi) [reply tests/test_l2_basic.py]: Added a "draft" version of create_lag API, please review

Comment (yuriilisovskyi) [reply tests/test_l2_basic.py]: Added a "draft" version of create_lag API with attrs=[] by default which simplifies the LAG creation and management. Please review

File: tests/test_l2_basic.py [hunk 7]
```diff
@@ -383,6 +383,183 @@ def test_l2_lag(npu, dataplane):
...
+
+            # Sending 100 packets to verify the order/sequence of distribution
+            for i in range(0, max_itrs):
+                src_mac = src_mac_start + str(i % 99).zfill(2)
+                ip_src = ip_src_start + str(i % 99).zfill(3)
+                ip_dst = ip_dst_start + str(i % 99).zfill(3)    
+                                
+                pkt = simple_tcp_packet(eth_dst=mac,
+                                        eth_src=src_mac,
+					                    ip_dst=ip_dst,
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:455]: looks like a mix of spaces an tabs

File: tests/test_l2_basic.py [hunk 8]
```diff
@@ -383,6 +383,183 @@ def test_l2_lag(npu, dataplane):
...
+                                        ip_src=ip_src,
+                                        tcp_sport=sport,
+					                    tcp_dport=dport,
```
Comment (andriy-kokhan) [inline tests/test_l2_basic.py:458]: looks like a mix of spaces an tabs

**Review by andriy-kokhan — APPROVED** [review]

**Pull request merged**